src/util/CartesianUtils.ts

```diff
@@ -32,15 +32,43 @@ export const formatAxisMap = (props: any, axisMap: any, offset: any, axisType: A
     const { orientation, domain, padding = {}, mirror, reversed } = axis;
     const offsetKey = `${orientation}${mirror ? 'Mirror' : ''}`;
 
-    let range, x, y, needSpace;
+    let calculatedPadding, range, x, y, needSpace;
+
+    if (axis.padding === 'gap' || axis.padding === 'no-gap') {
+      const diff = domain[1] - domain[0];
+      let smallestDistanceBetweenValues = Infinity;
+      const sortedValues = axis.categoricalDomain.sort();
+      sortedValues.forEach((value: number, index: number) => {
+        smallestDistanceBetweenValues = Math.min(
+          (value || 0) - (sortedValues[index - 1] || 0),
+          smallestDistanceBetweenValues,
+        );
+      });
+      const smallestDistanceInPercent = smallestDistanceBetweenValues / diff;
+      const rangeWidth = axis.layout === 'vertical' ? offset.top + offset.height : offset.width + offset.left;
+
+      if (axis.padding === 'gap') {
+        calculatedPadding = (smallestDistanceInPercent * rangeWidth) / 2;
+      }
+
+      if (axis.padding === 'no-gap') {
+        calculatedPadding = smallestDistanceInPercent * rangeWidth * 0.35;
```

Comment:
why is `0.35` ?

Reply:
refactored to be calculated correctly (0.35 was just a guess)